dive_core/command_hierarchy.cpp

```diff
@@ -75,6 +75,9 @@ uint64_t Topology::GetNumNodes() const
     DIVE_ASSERT(m_node_children.size() == m_node_shared_children.size());
     DIVE_ASSERT(m_node_children.size() == m_node_parent.size());
     DIVE_ASSERT(m_node_children.size() == m_node_child_index.size());
+    DIVE_ASSERT(m_node_children.size() == m_start_shared_child.size());
```

Comment:
why is m_node_children.size() == m_start_shared_child.size() == m_end_shared_child.size() == m_root_node_index.size()?

Reply:
ah, i see, it is because each non-root node will have 1 start, 1 end, and 1 root, right?

Reply:
not necessary in this change, but almost feel that it would be cleaner to have AoS instead of a SoA here.

Reply:
Every child has its own set of children, start/end child, parent, etc. So they all must be same size.
Yes, in some ways AoS is cleaner than SoA, especially when we have so many parallel arrays. But as with all things AoS, it's easy to get unwieldy and bolt on abstractions when you do AoS, and more importantly (as you know), it's likely less cache friendly.